pkg/slo-controller/noderesource/plugins/gpudeviceresource/plugin.go

```diff
@@ -161,14 +161,17 @@ func (p *Plugin) calculate(node *corev1.Node, device *schedulingv1alpha1.Device)
 	// calculate gpu resources
 	gpuResources := make(corev1.ResourceList)
 	totalKoordGPU := resource.NewQuantity(0, resource.DecimalSI)
+	healthGPUNum := 0
 	for _, d := range device.Spec.Devices {
 		if d.Type != schedulingv1alpha1.GPU || !d.Health {
 			continue
 		}
+		healthGPUNum++
 		util.AddResourceList(gpuResources, d.Resources)
 		totalKoordGPU.Add(d.Resources[extension.ResourceGPUCore])
 	}
 	gpuResources[extension.ResourceGPU] = *totalKoordGPU
+	gpuResources[extension.ResourceGPUShared] = *resource.NewQuantity(int64(healthGPUNum)*100, resource.DecimalSI)
```

Comment:
In some scenarios, the resource gpu.shared will be reported by DP. In this case, there may be conflicts between what DP reports and what is synchronized by koord-manager. Can we consider adding a featuregate to the synchronization of this resource?

Reply:
I have had add a  featuregate useDeviceGPUSharedResource  in colocationCfg, default is false which use shared 
 resource calculated by koord-manager. If true use shared resource reported by  DP reports.

Reply:
Rather than putting it in colocationCfg, it might be better to use featuregate under pkg/feature.

Reply:
add EnableDPGPUSharedResource in defaultFeatureGate